File: cmk/gui/userdb/type_defs.py
#!/usr/bin/env python3
# Copyright (C) 2023 Checkmk GmbH - License: GNU General Public License v2
# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
# conditions defined in the file COPYING, which is part of this source code package.

from typing import NamedTuple


class RelayState(NamedTuple):
    """SAML2 HTTP parameter"""

    target_url: str
    connection_id: str

    def __str__(self) -> str:
        return f"{self.connection_id},{self.target_url}"

File: packages/cmk-frontend-vue/src/lib/main-menu/service/type-defs.ts
/**
 * Copyright (C) 2025 Checkmk GmbH - License: GNU General Public License v2
 * This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
 * conditions defined in the file COPYING, which is part of this source code package.
 */
import type { NavItem, NavItemTopic } from 'cmk-shared-typing/typescript/main_menu'

import type { Colors } from '@/components/CmkBadge.vue'

export type OnNavigateCallback = (item: NavItem) => void
export type OnCloseCallback = (id: string) => void
export type OnToggleShowMoreLess = (id: string, show_more_active: boolean) => void
export type OnShowAllTopic = (id: string, topic: NavItemTopic) => void

export interface AjaxResponse<T> {
  result_code: number
  result: T
  severity: 'success' | 'error'
}
/**
 *  AjaxResults
 */
export interface UnackIncompWerksResult {
  count: number
  text: string
  tooltip: string
}

export interface UserMessagesResult {
  popup_messages: UserPopupMessage[]
  hint_messages: UserHintMessages
}

/**
 * Misc
 */

export interface UserPopupMessage {
  id: string
  text: string
  title?: string
}

export interface UserHintMessages {
  type: string
  title: string
  text: string
  count: number
}

export interface MenuItemBadge {
  content: string
  color: Colors
}
